packages/@lwc/synthetic-shadow/src/shared/handled-events.ts

```diff
@@ -0,0 +1,25 @@
+/*
+ * Copyright (c) 2018, salesforce.com, inc.
+ * All rights reserved.
+ * SPDX-License-Identifier: MIT
+ * For full license text, see the LICENSE file in the repo root or https://opensource.org/licenses/MIT
+ */
+
+const currentlyHandledEvents: Set<Event> = new Set();
+
+export function isBeingHandledByWrappedListener(event: Event): boolean {
+    return currentlyHandledEvents.has(event);
+}
+
+export function invokeWrappedListener(
+    listener: EventListener,
+    target: EventTarget | EventListenerObject,
+    event: Event
+): void {
+    currentlyHandledEvents.add(event);
+    try {
+        listener.call(target, event);
+    } finally {
+        currentlyHandledEvents.delete(event);
```

Comment:
this is missing a critical part, the throw! if the listener fails, you have the obligation to throw the even again. Look at the boundary protection for reference.

Reply:
try-finally without a catch block will still throw the exception. It will not swallow it.

Reply:
ah! you're right! we only need the catch if we want to inspect the error at this point, great!